src/main/scala/no/gdl/bookapi/controller/OPDSController.scala

```diff
@@ -8,56 +8,131 @@
     }
 
-    get("/:lang/catalog.atom") {
-      showOpdsForLanguage(params("lang"))
+    get(BookApiProperties.OpdsNavUrl.url) {
+      val navFeeds = feedService.feedsForNavigation(language("lang"))
+      val navUpdated = navFeeds.map(_.updated).sorted.reverse.headOption
+      navigationFeed(feedUpdated = navUpdated, feeds = navFeeds)
+    }
+
+    get(BookApiProperties.OpdsRootUrl.url) {
+      acquisitionFeed(books = feedService.allEntries(language("lang")))
+    }
+
+    get(BookApiProperties.OpdsNewUrl.url) {
+      acquisitionFeed(books = feedService.newEntries(language("lang")))
     }
 
-    get("/catalog.atom") {
-      showOpdsForLanguage(DefaultLanguage)
+    get(BookApiProperties.OpdsFeaturedUrl.url) {
+      val lang = language("lang")
+      acquisitionFeed(
+        feedUpdated = feedService.editorsPickLastUpdated(lang),
+        books = feedService.editorsPicks(lang))
     }
 
-    private def showOpdsForLanguage(language: String) = {
-      <feed xmlns="http://www.w3.org/2005/Atom" xmlns:dc="http://purl.org/dc/terms/" xmlns:opds="http://opds-spec.org/2010/catalog">
-        <id>http://api.digitallibrary.io/book-api/opds/catalog.atom</id>
-        <title>Global Digital Library - Open Access</title>
-        <updated>2017-04-28T12:54:15Z</updated>
-        <link href="http://api.digitallibrary.io/book-api/opds/catalog.atom" rel="self"/>
-        {readService.withLanguage(language, pageSize, page).results.map(x =>
-        <entry>
-          <id>urn:uuid:{x.uuid}</id>
-          <title>{x.title}</title>
-          {x.contributors.map(contrib => <author><name>{contrib.name}</name></author>)}
-          <updated>{sdf.format(new Date())}</updated>
-          <summary>{x.description}</summary>
-          {if(x.coverPhoto.isDefined) {
-            <link href={x.coverPhoto.get.large} type="image/jpeg" rel="http://opds-spec.org/image"/>
-            <link href={x.coverPhoto.get.small} type="image/png" rel="http://opds-spec.org/image/thumbnail"/>
-          }}
-          <link href={x.downloads.epub} type="application/epub+zip" rel="http://opds-spec.org/acquisition/open-access"/>
-          <link href={x.downloads.pdf} type="application/pdf" rel="http://opds-spec.org/acquisition/open-access"/>
-        </entry>
-      )}
-      </feed>
+    get(BookApiProperties.OpdsLevelUrl.url) {
+      acquisitionFeed(
+        titleArgs = Seq(params("lev")),
+        books = feedService.entriesForLanguageAndLevel(language("lang"), params("lev")))
     }
 
+    private def navigationFeed(feedUpdated: Option[LocalDate], feeds: => Seq[Feed])(implicit request: HttpServletRequest) = {
+      val lang = language("lang")
+      val selfOpt = feedService.feedForUrl(request.getRequestURI, lang, feedUpdated, Seq(), Seq())
+      selfOpt match {
+        case None =>
+          contentType = "text/plain"
+          NotFound(body = s"No books available for language $lang.")
+
+        case Some(self) => {
+          <feed xmlns="http://www.w3.org/2005/Atom">
```

Comment:
Should we have a unit test for the xml generated here and also in `acquisitionFeed`?